app/javascript/components/dialog-editor/TagControl.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import { componentTypes } from '@data-driven-forms/react-form-renderer';
+import { formFieldsMapper } from '@data-driven-forms/pf3-component-mapper';
+
+import { API } from '../../http_api';
+
+const Component = formFieldsMapper[componentTypes.SELECT];
+const url = id => `/api/categories/${id}`;
+
+export const RawTagControl = ({ categoryId, forceSingleValue, ...props }) => {
+  const [{ multi, loadOptions }, setState] = useState({
+    multi: true,
+    loadOptions: () => new Promise(() => undefined),
```

Comment:
It should be always resolved not return undefined. Also i don't think this should be in state at all. just create a helper function which will call correct API endpoint.

Reply:
Well, what if there's no category set in the editor and the user previews the dialog? I don't think that the promise should be resolved if there are no data provided...

Reply:
You have to resolve the prosmis otherwise the component will wait for ever. You can have some dummy

```jsx
new Promise(res => res(undefined))
```

Reply:
I exactly want it to wait forever, until the user sets the `catalogId`

Reply:
Oh ok then. Can the user pic the category only in the dialog editor or in the player?

Reply:
Only the editor, in the player it's fixed.

Reply:
So would not it be better to have this in the editor wrapper and send down the "real" props?

Reply:
I don't really understand this, what are the "real" props. I have to load the tags and the singleValue from the API anyway. Those can change while MiQ is running so we need them dynamically. The prop `categoryId` can not change in preview, but the contents the API requests might return different stuff for the same `categoryId`